exporters/provider_common/openshift.py

```diff
@@ -19,3 +69,234 @@ def convert_datetime(dt: Union[str, datetime]) -> datetime:
+        return [
+            ocp_object
+            for ocp_object in pods
+            if ocp_object.metadata.ownerReferences
+            and any(
+                owner_ref.kind in SUPPORTED_REPLICA_OBJECTS
+                for owner_ref in ocp_object.metadata.ownerReferences
+            )
+        ]
+
+    return pods
+
+
+def get_owner_object_from_child(
+    client: DynamicClient, uid: str, child_object: ResourceField
+) -> Dict[str, ResourceInstance]:
+    """
+    Retrieves the OpenShift Parent object by its UID, using information about the API version and resource type
+    from the given Child object.
+
+    Args:
+        client (DynamicClient): An OpenShift client object.
+        uid (str): The UID of the Parent object.
+        child_object (ResourceField): The Child object that contains the reference to the Parent object.
+
+    Returns:
+        Dict[str, ResourceField]: A dictionary with the UID of the Parent object as the key
+                                  and the Parent object itself as the value.
+                                  An empty dictionary is returned if the Parent object is not found
+                                  or if there is an error during the retrieval.
+    """
+
+    owner_ref = next(
+        (owner for owner in child_object.metadata.ownerReferences if owner.uid == uid),
+        None,
+    )
+
+    if owner_ref:
+        _remove_expired_objects()
+        replica = _get_object_from_cache(owner_ref.uid)
+        if replica:
+            return {owner_ref.uid: replica}
+
+        logging.debug(
+            "Getting replica: %s, kind: %s, api_version: %s",
+            owner_ref.uid,
+            owner_ref.kind,
+            owner_ref.apiVersion,
+        )
+
+        try:
+            api_resource = client.resources.get(
+                api_version=owner_ref.apiVersion, kind=owner_ref.kind
+            )
+
+            # We don't need to limit for a given namespace, because Replica objects may live in other
+            # then Pod namespace, so we may get multiple replica objects for a given name.
+            # The field_selector does not work on the UID, that's why we need to match separately
+            replica_list = api_resource.get(
+                field_selector=f"metadata.name={owner_ref.name}"
+            )
+
+            for replica in replica_list.items:
+                if replica.metadata.uid == owner_ref.uid:
+                    _add_object_to_cache(owner_ref.uid, replica)
+                    return {owner_ref.uid: replica}
+        except ResourceNotFoundError:
+            logging.debug(
+                "API Object not found for version: %s object: %s",
+                owner_ref.apiVersion,
+                owner_ref.uid,
+            )
+    return {}
+
+
+def filter_pods_by_replica_uid(
+    pods_list: List[ResourceField],
+) -> Dict[str, ResourceField]:
+    """
+    Filters out the given list of Pod objects to create a dictionary with ReplicaSet or
```

Comment:
I think the explanation here is too big and not clear

I understand as "return a relation 1-1 replica object uid to pods. Not all pod may appear in this ralation", am I right?

Reply:
The relation here is a bit more complex. In some cases it's many-many.

One pod may have multiple replica objects similarly one replica object may contain multiple pods.

The relation we are taking here is that we always have homogeneous pods within one replica.

In the scenario where we have multiple replicas and the same 2 pods controlled by those replicas, so:

```
replicaA --- podX
replicaA --- podY
replicaB --- podX
replicaB --- podY
replicaC --- podX
replicaC --- podY
```

We will get actually 3 deployments as our replica UID makes a `deployment` and here we have 3 different ReplicaSets controlling the same pods.

In the other scenario where we have two replicas controlling different pods:
```
replicaA --- podA
replicaA --- podB
replicaA --- podC
replicaB --- podX
replicaB --- podY
replicaB --- podZ
```

We will get 2 deployments as we have 2 replica unique UIDs.

Reply:
Adding an example to the description seems like a good solution

But I would also want tests for it

for example, for the first scenario, this are both valid returns from the function, right?
```py
# option 1
{'uidA': <podX>, 'uidB': <podX>, 'uidC': <podX>}
# option 2
{'uidA': <podY>, 'uidB': <podY>, 'uidC': <podY>}
# there are more options still, right?
``` 

and for the second scenario
```py
# option 1
{'uidA': <podA>, 'uidB': <podX>}
# option 2
{'uidA': <podB>, 'uidB': <podY>}
# there are more options still, right?
``` 

since the return of the function can be not unique, this can lead us to not collect all data?